### DISCUSSION TITLE
Allow using prefixed relative paths in `metadata`, especially in `alternates.languages`

### DISCUSSION THREAD

**Discussion by cprussin** [vercel/next.js#61827] [Ideas]
### Goals

Allow syntax like this:

```ts
export const metadata = {
  metadataBase = new URL("http://example.com"),
  alternates: {
    canonical: "./",
    languages: {
      es: "/es/./",
      fr: "/fr/./"
    }
  }
};
```

With this syntax, `es` would then expand to `http://example.com/es/the/current/pathname`.

### Background

This would enable configuring alternates in a layout without having to propagate the configuration to each page file and manually expand pathnames.  Without such syntax it's nearly impossible to reliably configure alternates across a large app  since doing so would require that developers remember to inject the metadata on every page they create which is super error prone.

### Proposal

In https://github.com/vercel/next.js/blob/e838f268ece3eed20c46a7ba168999c5c5af208b/packages/next/src/lib/metadata/resolvers/resolve-url.ts#L75-L80, we could change the function body to something like:

```ts
function resolveRelativeUrl(url: string | URL, pathname: string): string | URL {
  if (typeof url === 'string') {
    const match = /(^|.*\/)(\.\/.*)/.exec(url)
    if (match) {
      return path.posix.resolve(match[1] + pathname, match[2])
    }
  }
  return url
}
```

And yes I'm happy to contribute if this contribution would be accepted.

**Comment by cprussin** [comment]
@huozhi @shuding  @timneutkens from commit logs it looks like you three are the most active folks working on metadata who have approval / merge access -- if one of you can give me a sign-off on accepting this feature I'll happily put in a PR to implement it!

**Comment by lednhatkhanh** [comment]
+1 for this as well!

**Comment by neo-elgiz-abbasov** [comment]
+1 here 🥺

**Comment by eason-dev** [comment]
+1 need this!
